smartsnippets_inherit/static/admin/smartsnippet_inherit/js/plugininherit_form.js

```diff
@@ -0,0 +1,205 @@
+(function($) {
+
+    _editor = {
+        _initial: {},
+        _variablesSubmission: false,
+        getInfoBox: function(text, type){
+            return "<span class='ssvar-info " + (type? type: '') + "'>" + text + "</span>"
+        },
+        getVariablesEl: function(){
+            return $(this.options.variables)
+                .find("[name^='_'][name$='_'][id^='var_']");
+        },
+        getVariableLabel: function(varEl){
+            var varsBox = $(this.options.variables);
+            return (
+                varsBox.find('label[for="' + varEl.attr('id') + '"]').text() ||
+                varEl.attr('name').replace(/(^_)|(_$)/g, '')
+            );
+        },
+        variablesToString: function(varsEl){
+            var self = this;
+            return $.map(varsEl, function(el){
+                return self.getVariableLabel($(el));
+            }).join(', ');
+        },
+        registerInitialData: function(snippetId){
+            var self = this;
+            self._initial[snippetId] = {};
+            self.getVariablesEl().each(function(){
+                self._initial[snippetId][$(this).attr('name')] = $(this).val();
+            });
+        },
+        submitOverwriteData: function(snippetId){
+            var self = this;
+            return function() {
+                var varsBox = $(self.options.variables);
+                if (!SnippetWidgetRegistry.allValid()){
```

Comment:
SnippetWidgetRegistry is something we handle and in my opinion it's the developer's responsibility to include the required js libs for this to work. If SnippetWidgetRegistry is not defined I do expect it to throw an error just so we get notified that the required lib is not available.